server/src/main/java/org/opensearch/common/cache/store/OpenSearchOnHeapCache.java

```diff
@@ -25,58 +33,66 @@
  *
  * @opensearch.experimental
  */
-public class OpenSearchOnHeapCache<K, V> implements StoreAwareCache<K, V>, RemovalListener<K, V> {
-
-    private final Cache<K, V> cache;
+public class OpenSearchOnHeapCache<K, V> implements ICache<K, V>, RemovalListener<ICacheKey<K>, V> {
 
-    private final StoreAwareCacheEventListener<K, V> eventListener;
+    private final Cache<ICacheKey<K>, V> cache;
 
-    private final CacheStats stats = new OpenSearchOnHeapCacheStats();
+    private final CacheStats stats;
+    private final RemovalListener<ICacheKey<K>, V> removalListener;
 
     public OpenSearchOnHeapCache(Builder<K, V> builder) {
-        CacheBuilder<K, V> cacheBuilder = CacheBuilder.<K, V>builder()
+        CacheBuilder<ICacheKey<K>, V> cacheBuilder = CacheBuilder.<ICacheKey<K>, V>builder()
             .setMaximumWeight(builder.getMaxWeightInBytes())
             .weigher(builder.getWeigher())
             .removalListener(this);
         if (builder.getExpireAfterAcess() != null) {
             cacheBuilder.setExpireAfterAccess(builder.getExpireAfterAcess());
         }
         cache = cacheBuilder.build();
-        this.eventListener = builder.getEventListener();
+        String dimensionName = Objects.requireNonNull(builder.shardIdDimensionName, "Shard id dimension name can't be null");
```

Comment:
This cache shouldn't be limited to a specific dimension. It is used by IRC this way but we need to make it generic(in terms of dimension) going forward. Same with other caches.